src/components/file/File.js

```diff
@@ -27,47 +28,55 @@ export class File extends React.Component {
     progress: PropTypes.number,
     error: PropTypes.string,
     isDownload: PropTypes.bool,
-    doGetDownloadLink: PropTypes.func
+    doDownloadFile: PropTypes.func
   }
 
-  state = {
-    progress: null
+  handleDownloadClick = async () => {
+    const { id, hash, name, doDownloadFile } = this.props
+    const file = await doDownloadFile(id, hash)
```

Comment:
Thinking out loud, and for another PR, perhaps we should start fetching the files from IPFS eagerly, before the user clicks on them.

Reply:
That would be a nice optimisation 🤔

I'll keep this in mind and maybe open an issue about it later!